agenda-connectors-services/src/main/java/org/exoplatform/agendaconnector/service/ExchangeConnectorServiceImpl.java

```diff
@@ -134,5 +145,46 @@ private ExchangeService connectExchangeServer(ExchangeService exchangeService,
     exchangeService.setUrl(new URI(exchangeServerURL + ExchangeConnectorUtils.EWS_URL));
     exchangeService.getInboxRules();
     return exchangeService;
+
   }
+
+  @Override
+  public void pushEventToExchange(long identityId, EventEntity event, ZoneId userTimeZone) throws IllegalAccessException {
+    ExchangeUserSetting exchangeUserSetting = getExchangeSetting(identityId);
+
+    try (ExchangeService exchangeService = new ExchangeService(ExchangeVersion.Exchange2010_SP2)) {
+      connectExchangeServer(exchangeService, exchangeUserSetting);
+      RemoteEvent remoteEvent = agendaRemoteEventService.findRemoteEvent(event.getId(), identityId);
+      if(remoteEvent == null){
+        Appointment meeting = new Appointment(exchangeService);
+        meeting.setSubject(event.getSummary());
+        ZonedDateTime startDate = AgendaDateUtils.parseRFC3339ToZonedDateTime(event.getStart(), userTimeZone);
+        ZonedDateTime endDate = AgendaDateUtils.parseRFC3339ToZonedDateTime(event.getEnd(), userTimeZone);
+        meeting.setStart(AgendaDateUtils.toDate(startDate));
+        meeting.setEnd(AgendaDateUtils.toDate(endDate));
+        meeting.save(new FolderId(WellKnownFolderName.Calendar), SendInvitationsMode.SendToAllAndSaveCopy);
+        RemoteEvent remoteEvent1 = new RemoteEvent();
+        remoteEvent1.setIdentityId(identityId);
+        remoteEvent1.setEventId(event.getId());
+        remoteEvent1.setRemoteProviderId(event.getRemoteProviderId());
+        remoteEvent1.setRemoteProviderName(event.getRemoteProviderName());
+        remoteEvent1.setRemoteId(String.valueOf(meeting.getId()));
+        agendaRemoteEventService.saveRemoteEvent(remoteEvent1);
+      }else{
+        ItemId itemId = new ItemId(remoteEvent.getRemoteId());
+        Appointment appointment = Appointment.bind(exchangeService,itemId);
+        appointment.setSubject(event.getSummary());
+        ZonedDateTime startDate = AgendaDateUtils.parseRFC3339ToZonedDateTime(event.getStart(), userTimeZone);
+        ZonedDateTime endDate = AgendaDateUtils.parseRFC3339ToZonedDateTime(event.getEnd(), userTimeZone);
+        appointment.setStart(AgendaDateUtils.toDate(startDate));
+        appointment.setEnd(AgendaDateUtils.toDate(endDate));
+        appointment.update(ConflictResolutionMode.AlwaysOverwrite,  SendInvitationsOrCancellationsMode.SendToAllAndSaveCopy);
+      }
+      // Verify that the meeting was created.
+    } catch (ServiceLocalException e) {
+      throw new IllegalAccessException("User " + identityId + " is not allowed to push exchange event informations");
+    } catch (Exception e) {
+      throw new IllegalAccessException("User " + identityId + " is not allowed to connect to exchange server");
+    }
+    }
```

Comment:
```suggestion
    } catch (ServiceLocalException e) {
      throw new IllegalAccessException("User '" + identityId + "' is not allowed to push exchange event informations");
    } catch (Exception e) {
      throw new IllegalAccessException("User '" + identityId + "' is not allowed to connect to exchange server");
    }
  }
```
Update also with the same way the exceptions log messages for getExchangeEvents method